### ISSUE TITLE
tavern.core.run selecting test file using pytest "-k EXPRESSION" option

### ISSUE THREAD

**Issue by pmull** [taverntesting/tavern#221]
When you call "tavern.core.run" and supply a test file to run, you would expect it to run only that test file; in essence, call "pytest <file>" on that file. However, how this is implemented is it is calling "pytest -k <file>", which has the negative effect of scanning for all "test_" files in the directory tree, loading them, and then deselecting them if they don't match the name of the file. If any of the files fails to load, then pytest will crash with error. Also, this is inefficient.

### Example
I have 2 files in the same directory, each with 1 test:

`test_tavern.tavern.yaml   test_tavern2.tavern.yaml`

I have a python file "t.py":

```
from tavern.core import run
run('test_tavern.tavern.yaml')
```

If they both load, then you get:

```
...
================= 1 passed, 1 deselected, 2 warnings in 0.27 seconds =================
```

However, if test_tavern2.tavern.yaml fails to load due to include file problem, then you get

```...
E   IOError: [Errno 2] No such file or directory: u'/Users/pmullane/files/workspace/test/common.yaml'
!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!!!!
============================== 1 error in 0.29 seconds ===============================
```

Calling pytest directly on the intended file works as expected.

```
...
======================== 1 passed, 1 warnings in 0.27 seconds ========================
```

**Comment by michaelboulton** [comment]
Thanks for spotting this - this was a while ago, but I think it was done to selectively choose files from sub directories easier before we had any kind of tagging on tests. This might change the behaviour of test runs for some people, but the way it is working now is actually contrary to the documentation (https://taverntesting.github.io/documentation.html#using-the-run-function)

**Comment by michaelboulton** [comment]
Changing it to remove `-k` from the call speeds up the integration tests quite a bit as well

**Comment by michaelboulton** [comment]
This should be fixed in version 0.22.0

**Comment by pmull** [comment]
Thanks much for the quick fix!
